### ISSUE TITLE
Changes to the Matrix rooms following the Libera.chat decision

### ISSUE THREAD

**Issue by GregSutcliffe** [ansible-community/community-topics#258]
### Summary

**This is informational**. No vote is needed, and I will be doing the work, I'm just posting this so that people are aware.

Following the [decision by Libera.chat](https://libera.chat/news/matrix-deportalling) to disallow portalled rooms from Matrix (see also the [response from Matrix.org here](https://matrix.org/blog/2023/07/deportalling-libera-chat/), we will need to make changes in our chat space. 

Sadly, pretty much every room we have today with an IRC bridge will be affected, and [the current advice](https://matrix.org/blog/2023/07/make-sure-libera-bridge-keeps-working/) is to create *new* Matrix rooms that are configured correctly.

The deadline is 31st July, so next week I'll be working through creating all the new rooms. IRC users should see no effect, Matrix users in a given room will be shown a message ("this room has been replaced, the conversation continues [here](url)") and once they click the "here" link they'll be moved over to the new room. They'll be unable to post in the old room at that point, so no history should be lost.

It's unfortunate that this is necessary, but hopefully the impact will be minimal. Please do ask any questions you may have, and I'll do my best to answer them.

TLDR - Matrix users should look out for this in rooms they participate in.

**Comment by samccann** [comment]
Discussed in the meeting. Some confusion on the matrix user impact etc but folks are aware this is happening by the end of the month.

**Comment by GregSutcliffe** [comment]
@samccann can you elaborate? I'm happy to inform people further.

**Comment by samccann** [comment]
I think it had to do with whether someone coming into the old room after  the move will have a clear message there to say this room is old and go to xx.  So like the last message in the room before it's read-only or whatever.

The info at the top of a room is kind of subtle and might not be noticed by many matrix users.

**Comment by GregSutcliffe** [comment]
Understood. Yes, the announcements are clear - or at least they are on Element (this is a client-side thing, so other clients may vary). The text-entry box for your next message gets replaced with a statement and a link to click to move to the new room, like this:

![2023-07-24-093650_455x46_scrot](https://github.com/ansible-community/community-topics/assets/827786/0cff4498-098f-4dc1-9261-c6d8073e115a)

**Comment by mariolenz** [comment]
@GregSutcliffe I think it might be useful to track what rooms you've already changed and what rooms you didn't touch yet. How about having a list like:

- [ ] room 1
- [ ] room 2
- [ ] room 3

and you tick off the rooms you've finished with? This might help people to check if they're still in the correct room or not in case their client doesn't show the message that it's been moved. I guess you have a ToDo list of rooms to work on, anyway, so why not make it public?

What do you think?

**Comment by GregSutcliffe** [comment]
Good idea! Here's the full list - caveat that not all rooms have bridges, and I'm not joining every room ahead of time to check, I'll just skip/mark-done those rooms as I get to them :)

In size (members) order:

- [ ] Ansible Suisse
- [ ] Ansible Linode
- [ ] Ansible DigitalOcean
- [ ] Ansible Español
- [ ] Ansible Storage
- [ ] Ansible JBoss
- [ ] Ansible India
- [ ] Ansible Lockdown
- [ ] Breakout 3
- [ ] Breakout 4
- [ ] Breakout 2
- [ ] Ansible Azure
- [ ] Ansible MySQL
- [ ] Ansible Container
- [ ] Breakout 1
- [ ] Ansible Security Automation
- [ ] Ansible Edge
- [ ] Ansible Zuul
- [ ] Ansible Website WG
- [ ] Ansible Diversity
- [ ] Ansible Meetings
- [ ] Ansible Workshops
- [ ] Ansible Kubernetes
- [ ] Ansible Europe
- [ ] Ansible Packaging
- [ ] ARA Records Ansible
- [ ] Ansible VMware
- [ ] Event Driven Ansible
- [ ] Ansible en Français
- [ ] community-zabbix/community
- [ ] Ansible Windows
- [ ] Ansible + AWS
- [ ] Ansible PostgreSQL
- [ ] Ansible Galaxy
- [ ] Ansible Contributor Summit
- [ ] AAP Configuration as Code
- [ ] Ansible Network
- [ ] Ansible Docs
- [ ] Ansible DevTools
- [ ] Ansible Development
- [ ] Ansible AWX
- [ ] Ansible Community Working Group
- [ ] Ansible Social
- [ ] Ansible Lightspeed
- [ ] Ansible User Help
- [ ] ansible/ansible

**Comment by GregSutcliffe** [comment]
Just a heads-up - the plumbing tools were quite unstable last week, so progress was slow. This affected more than just us, so Matrix & Libera have agreed a 2 week extension. Details [here](https://matrix.org/blog/2023/07/postponing-libera-chat-deportalling/) and [here](https://libera.chat/news/deportalling-delay). New portal creation will be disabled today, but our existing channels should continue working until the 11th Aug.

I'm going to be at Flock this week, so there is time for the work on the tooling to be done. I will resume room migration next weekend. Thanks for your patience everyone!

**Comment by GregSutcliffe** [comment]
Hi all,

As you may have seen, the IRC is currently down, and we have no ETA for it's return. Given the [issues discussed by Libera](https://libera.chat/news/temporarily-disabling-the-matrix-bridge) it won't be "days", but "weeks", I would imagine.

This is obviously painful for us. We are not in a place where we really want to discard either half of the community (and it is very close to 50/50, although Matrix continues to grow slowly). As a temporary and *very* rough bandage on the problem, I have deployed [Matterbridge](https://github.com/42wim/matterbridge/) on the community infra to provide a relaybot between IRC and Matrix **in the #docs room only**. If that seems ok in the next day or so, I can roll it out in other rooms - please let me know if there is a room you'd like set up this way.

Let's hope this is **really** temporary. Thanks for your patience!
